tests/steps/backend/reservation/create_reservation_steps.py

```diff
@@ -0,0 +1,48 @@
+from assertpy import assert_that
+from behave import given, when, then
+
+from actions.api.reservation_endpoint_actions import do_post_request_to_create_reservation
+from models.endpoints.reservations_model import get_valid_create_reservation_payload
+from tests.steps.backend.book.create_book_steps import given_i_have_a_new_book_added_into_database
+from tests.steps.backend.user.get_users_steps import given_i_have_a_new_user_added_into_database
+
+
+@given('I have a new user and book added into database')
+def given_i_have_a_new_user_and_book_added_into_database(context):
+    given_i_have_a_new_user_added_into_database(context)
+    context.user_id = context.response.json()['id']
```

Comment:
make a response local variable since you use context.response.json() so many times